### PR TITLE
feat: add context parameter to POST /verify

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

Add optional `context` parameter to `POST /verify` for passing request-scoped attributes to ABAC policy evaluation.

- Add `requestContext` to `CollectorContext` type
- Create `RequestContextCollector` — extracts client IP from `context.ip`
- Update verify route to read `context` from request body
- Export `RequestContextCollector` and `ATTR_CLIENT_IP`
- Backward compatible: `context` is optional

Replaces the removed `ip` claim from JWT token payload (standardized in auth.provider#20).

## Test plan

- [x] 126 tests pass (28 test files)
- [x] RequestContextCollector extracts ip from context
- [x] Returns empty for missing/non-string/absent ip
- [x] Verify route passes context to collectors
- [x] Verify route works without context (backward compatible)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/server/routes/verify.mts [hunk 1]
```diff
@@ -47,11 +47,11 @@ export function createVerifyRouter(config: VerifyRouterConfig): express.Router {
 				tokenType,
 			};
 
-			const { resource: rawResource, action } = req.body;
+			const { resource: rawResource, action, context: requestContext } = req.body;
 			const resource = config.resourceParser.parse(rawResource);
 			const requestId = req.get("x-request-id");
 			const headers = requestId ? { "x-request-id": requestId } : undefined;
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/server/routes/verify.mts:53]: By design — verify API is internal, called by services (grpc.authz, etc.), not by end users. The service reads req.ip from the original request and passes it in context. Network-level access control (service mesh / firewall) prevents direct end-user access to /verify.

File: src/collectors/RequestContextCollector.mts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import { ATTR_CLIENT_IP } from "#/engine/keys.mjs";
+import type { AttributeCollector, Attributes, CollectorContext } from "#/engine/types.mjs";
+
+export class RequestContextCollector implements AttributeCollector {
+	async collect(context: CollectorContext): Promise<Attributes> {
+		const attrs: Attributes = new Map();
+		const ip = context.requestContext?.ip;
+		if (typeof ip === "string") {
+			attrs.set(ATTR_CLIENT_IP, ip);
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/collectors/RequestContextCollector.mts]: Fixed in 20b45e0. Added trim() and empty string check. Skipped net.isIP — too restrictive for edge cases like IPv6 zone IDs.

File: src/server/__tests__/verify.test.mts [hunk 3]
```diff
@@ -25,6 +26,22 @@ function createTestApp(resourceParser?: ResourceParser) {
 	return app;
 }
 
+function createTestAppWithContext() {
+	const app = express();
+	app.use(
+		createVerifyRouter({
+			jwt: { secret: JWT_SECRET, validate: false },
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/server/__tests__/verify.test.mts]: Fixed in 20b45e0. Changed to validate: true.

File: src/server/__tests__/verify.test.mts [hunk 4]
```diff
@@ -68,6 +85,30 @@ describe("POST /verify", () => {
 		expect(res.body.code).toBe("invalid_token");
 	});
 
+	it("passes context to collectors", async () => {
+		const app = createTestAppWithContext();
+		const token = jwt.sign({ scope: "read:project" }, JWT_SECRET);
+		const res = await request(app)
+			.post("/verify")
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/server/__tests__/verify.test.mts]: Fixed in 2ab795d. Renamed test to "accepts context in request body without error" to accurately reflect what it asserts.

File: src/collectors/RequestContextCollector.mts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import { ATTR_CLIENT_IP } from "#/engine/keys.mjs";
+import type { AttributeCollector, Attributes, CollectorContext } from "#/engine/types.mjs";
+
+export class RequestContextCollector implements AttributeCollector {
+	async collect(context: CollectorContext): Promise<Attributes> {
+		const attrs: Attributes = new Map();
+		const ip = context.requestContext?.ip;
+		if (typeof ip === "string" && ip.trim()) {
+			attrs.set(ATTR_CLIENT_IP, ip.trim());
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/collectors/RequestContextCollector.mts]: Fixed in 2ab795d. Extracted to local variable, single trim() call.
